src/defaultsettings.cpp

```diff
@@ -39,11 +101,7 @@ void set_default_settings()
 	// Client
 	settings->setDefault("address", "");
 	settings->setDefault("enable_sound", "true");
-#if ENABLE_TOUCH
-	settings->setDefault("enable_touch", "true");
-#else
-	settings->setDefault("enable_touch", "false");
-#endif
+	settings->setDefault("enable_touch", bool_to_cstr(detect_touch()));
```

Comment:
`detect_touch` should be called once and the result should be stored in a local variable.